backend/src/routes/api/groups-config/groupsConfigUtil.ts

```diff
@@ -0,0 +1,192 @@
+      );
+    }
+    return {
+      success: updatedConfig.groupsConfig,
+      error: null,
+    };
+  } catch (e) {
+    fastify.log.error('Error updating group configuration' + e.toString());
+    const error = createError(500, 'Error updating group configuration');
+    throw error;
+  }
+};
+
+const processGroupData = (groupsData: GroupConfigMapData): GroupConfigMapDataProcessed => {
+  const adminGroupsList = groupsData.admin_groups.split(',');
+  const userGroupList = groupsData.allowed_groups.split(',');
+
+  return {
+    adminGroup: adminGroupsList,
+    allowedGroup: userGroupList,
+  };
+};
+
+const processGroupConfig = (
+  groupsDataProcessed: GroupConfigMapDataProcessed,
+  groups: string[],
+): { groupsConfig: GroupsConfig; groupsConfigMapData: GroupConfigMapData } => {
+  const adminGroupsConfig = groups.map((group, index) => ({
+    id: index,
+    name: group,
+    enabled: groupsDataProcessed.adminGroup.includes(group),
+  }));
+  const userGroupsConfig = groups.map((group, index) => ({
+    id: index,
+    name: group,
+    enabled: groupsDataProcessed.allowedGroup.includes(group),
+  }));
+
+  userGroupsConfig.push({
+    id: userGroupsConfig.length,
+    name: SYSTEM_AUTHENTICATED,
+    enabled: groupsDataProcessed.allowedGroup.includes(SYSTEM_AUTHENTICATED),
+  });
+
+  const groupsConfig: GroupsConfig = {
+    adminGroups: adminGroupsConfig,
+    userGroups: userGroupsConfig,
+  };
+
+  const missingAdminGroups = groupsDataProcessed.adminGroup.filter(
+    (group) => !groups.includes(group),
+  );
+  const missingUserGroups = groupsDataProcessed.allowedGroup.filter(
+    (group) => !groups.includes(group) && group !== SYSTEM_AUTHENTICATED,
+  );
+
+  if (missingAdminGroups.length === 1) {
+    groupsConfig.errorAdmin = `The group ${missingAdminGroups.join(
+      ', ',
+    )} no longer exists in OpenShift and has been removed from the selected group list.`;
+  } else if (missingAdminGroups.length > 1) {
+    groupsConfig.errorAdmin = `The groups ${missingAdminGroups.join(
+      ', ',
+    )} no longer exist in OpenShift and has been removed from the selected group list.`;
+  }
+
+  if (missingUserGroups.length === 1) {
+    groupsConfig.errorUser = `The group ${missingUserGroups.join(
+      ', ',
+    )} no longer exists in OpenShift and has been removed from the selected group list.`;
+  } else if (missingUserGroups.length > 1) {
+    groupsConfig.errorUser = `The groups ${missingUserGroups.join(
+      ', ',
+    )} no longer exist in OpenShift and has been removed from the selected group list.`;
+  }
+
+  const updatedConfigMap: GroupConfigMapData = {
+    admin_groups: groupsDataProcessed.adminGroup
+      .filter((group) => groups.includes(group))
+      .toString(),
+    allowed_groups: groupsDataProcessed.allowedGroup
+      .filter((group) => groups.includes(group) || group === SYSTEM_AUTHENTICATED)
+      .toString(),
+  };
+
+  return { groupsConfig: groupsConfig, groupsConfigMapData: updatedConfigMap };
```

Comment:
```suggestion
  return { groupsConfig, groupsConfigMapData: updatedConfigMap };
```

Reply:
Isn't it better to explicityly declare the key of the attribute?

Reply:
It's just short-hand syntax, which is more preferred as are many things these days... JS has gone through a history and a half of changes and QoL things. Promises & Async/Await, fetch vs XMLHttpRequest, etc, etc... 

`{ myKey: myValue }` is fine, no one bats an eye, name your variables whatever you want and key them up in a new object however you want. ie `{ groupsConfigMapData: updatedConfigMap }` is fine if you're cool with the key being named that.

`{ data: data }` is sorta odd to see in modern JS these days. Because one of the magical shorthands is if the `key` is named the same as your `variable`, you can just simplify it and save the extra few characters `{ data }` which naturally still is `{ data: data }` to the runtime (or depending on webpack settings, changed on output for the minified js).

Reply:
great, thank you very much andrew, so insightful. Changed!